app/code/Magento/InventoryGroupedProductAdminUi/Observer/ProcessSourceItemsObserver.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryGroupedProductAdminUi\Observer;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\GroupedProduct\Model\Product\Type\Grouped as GroupedProductType;
+use Magento\InventoryCatalogAdminUi\Observer\SourceItemsProcessor;
+
+/**
+ * Process source items for grouped associated products during product saving via controller.
+ */
+class ProcessSourceItemsObserver implements ObserverInterface
+{
+    /**
+     * @var SourceItemsProcessor
+     */
+    private $sourceItemsProcessor;
+
+    /**
+     * @param SourceItemsProcessor $sourceItemsProcessor
+     */
+    public function __construct(
+        SourceItemsProcessor $sourceItemsProcessor
+    ) {
+        $this->sourceItemsProcessor = $sourceItemsProcessor;
+    }
+
+    /**
+     * @param EventObserver $observer
+     * @return void
+     */
+    public function execute(EventObserver $observer): void
+    {
+        $product = $observer->getEvent()->getProduct();
```

Comment:
We should create a plugin on service which is responsible for assigning children to grouped product.
And need to run reindex after calling this service.

In current case reindexation will be run if you save product via admin panel but not via REST call
Also, it is overhead to save source items only for run indexer (pay attention we don't expect any changes in qty or stock status, it is impossible do via grouped product admin page)